app/controllers/api/v1/messages_controller.rb

```diff
@@ -2,31 +2,20 @@ class Api::V1::MessagesController < ApplicationController
   before_action :set_channel
 
   def index
-    messages = @channel.messages.order('created_at ASC').map do |message|
-      render_message(message)
-    end
-    render json: messages
+    messages = @channel.messages.order('created_at ASC')
+    render json: messages # see Message.as_json method
   end
 
   def create
     message = @channel.messages.build(content: params[:content])
     message.user = current_user
     message.save
-    render json: render_message(message)
+    render json: message # see Message.as_json method
   end
 
   private
 
   def set_channel
     @channel = Channel.find_by(name: params[:channel_id])
   end
-
-  def render_message(message)
```

Comment:
I moved this code to the `Message` model, in [`as_json`](http://api.rubyonrails.org/v5.1/classes/ActiveModel/Serializers/JSON.html#method-i-as_json).